### ISSUE TITLE
Invalid negative/zero values returned by Dimensions API under  iOS 15

### ISSUE THREAD

**Issue by beau1399** [facebook/react-native#32940]
### Description

Several of our users are reporting problems that we've found to be attributable to Dimensions.get('window') erroneously returning 0 for width. 

### Version

0.63.4

### Output of `npx react-native info`

info Fetching system and libraries information...
System:
    OS: macOS 12.1
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 1.89 GB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.17.3 - ~/qgenda.mobile/node_modules/.bin/yarn
    npm: 8.1.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
    Watchman: 2021.11.01.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /Users/beauwilkinson/.rbenv/shims/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.2, iOS 15.2, macOS 12.1, tvOS 15.2, watchOS 8.3
    Android SDK: Not Found
  IDEs:
    Android Studio: 2020.3 AI-203.7717.56.2031.7678000
    Xcode: 13.2/13C5066c - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_301 - /usr/bin/javac
    Python: 2.7.18 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.14.0 
    react-native: 0.63.4 => 0.62.3 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

Call Dimensions.get('window').width on an iPhone 13 Pro Max (iPhone14,3) running iOS 15. The value obtained will be 0.

**Comment by ewfian** [comment]
enable `Fast App Termination` and leave app on background. about 1h later, switch to foreground. the issue may appeared

**Comment by adamivancza** [comment]
I'm facing the same issue on 0.62.3 and I think I found out why is that happening. This is the piece of the relevant code that figures out screen size on 0.62.3: https://github.com/facebook/react-native/blob/83425fa72f1f9a12f6b6deefd66fa8bd154a3651/React/UIUtils/RCTUIUtils.m#L19
And this was changed on a later version to: https://github.com/facebook/react-native/blob/8bd3edec88148d0ab1f225d2119435681fbbba33/React/UIUtils/RCTUIUtils.m#L20 as part of this commit: https://github.com/facebook/react-native/commit/e85372298109abf258d5154e2a28bc6496fb9529

so after I moved this changed to 0.62.3 this issue was resolved for me.
Here is a patch file which you can run to fix this:

```
diff --git a/node_modules/react-native/React/UIUtils/RCTUIUtils.m b/node_modules/react-native/React/UIUtils/RCTUIUtils.m
index 64b97bf..d729310 100644
--- a/node_modules/react-native/React/UIUtils/RCTUIUtils.m
+++ b/node_modules/react-native/React/UIUtils/RCTUIUtils.m
@@ -16,7 +16,7 @@ RCTDimensions RCTGetDimensions(CGFloat fontScale)
 
   UIView *mainWindow;
   mainWindow = RCTKeyWindow();
-  CGSize windowSize = mainWindow.bounds.size;
+  CGSize windowSize = mainWindow ? mainWindow.bounds.size : screenSize;
 
   RCTDimensions result;
   typeof (result.screen) dimsScreen = {
```

**Comment by alkodrimuhfaris** [comment]
Is it confirmed as an issue? Because several users from my app also encounter this issue on iOS 15.3.1 we're using react-native 0.60.4

**Comment by adamivancza** [comment]
@alkodrimuhfaris yep - this was fixed 0.64.0

**Comment by alkodrimuhfaris** [comment]
@adamivancza Okay great! Thanks!
